projects/GKCore/GKCore/Controllers/TreeChartWinController.cs

```diff
@@ -131,17 +131,20 @@ private void ParentAdd(GDMSex needSex)
                 fBase.Context.Tree.GetSpouses(fam, out father, out mother);
 
                 needParent = (father == null && needSex == GDMSex.svMale) ||
-                    (mother == null && needSex == GDMSex.svFemale);
+                             (mother == null && needSex == GDMSex.svFemale);
+            } else {
+                needParent = true;
             }
 
-            if (!familyExist || needParent) {
-                GDMIndividualRecord child = p.Rec;
-                fam = (familyExist) ? fBase.Context.GetParentsFamily(p.Rec) : fBase.Context.Tree.CreateFamily();
-                GDMIndividualRecord parent = fBase.Context.SelectPerson(null, TargetMode.tmParent, needSex);
+            if (needParent) {
+                GDMIndividualRecord parent = fBase.Context.SelectPerson(p.Rec, TargetMode.tmChild, needSex);

```

Comment:
`TargetMode` means a type of target. Here the target is a child, so we send `tmChild`. Previously it did not matter because target was `null`.